google-beta/resource_dns_managed_zone.go

```diff
@@ -54,6 +55,34 @@ func resourceDnsManagedZone() *schema.Resource {
 				Optional: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+			"private_visibility_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"networks": {
+							Type:     schema.TypeList,
```

Comment:
Do you know if the networks always come back in the same order, and if that order is the same one that's in the config (vs, say, alphabetical)?

Reply:
The order shouldn't matter, but I couldn't find a way to have magic-modules generate a schema.TypeSet.

Reply:
ah ok, you can do it by putting `is_set: true` on the ResourceOverride. since it's a set of URLs though you would also need to do a bit more to make sure the hash is the same regardless of beta vs ga. Easiest way to do that is to also put `set_hash_func: selfLinkNameHash`

Reply:
I'm not quite sure how this would work. I can add `is_set: true` on `networks`, but `networks` is not a set of URLs. It is a set of NestedObjects each with a child `networkUrl` resource. Do I need a custom `set_hash_func` that will hash only the child `networkUrl` field?

Reply:
I considered changing `networks` into a set of URLs using a `custom_expand` and `custom_flatten` for Terraform only on terraform.yaml, so the definition would look something like this:

```
private_visibility_config {
  networks = ["${google_compute_network.network-1.self_link}",
      "${google_compute_network.network-2.self_link}"]
}
```

This almost works, but I cannot override the type of `networks` in terraform.yaml to be a `Api::Type::String` instead of `Api::Type::NestedObject`. And I wouldn't like to make this change in api.yaml as that should match the API definition.

Reply:
Oh, you're right. Yup, you would want a custom `set_hash_func` in that case.

Reply:
I've encountered this error before! When you call `d.Set` with a `[]interface{}`, Terraform will magic it into a top-level `TypeSet`. But when you nest a `TypeSet` inside a `TypeList` it doesn't, keeping it as a list. You'll need need to explicitly set the type and define the hash function. If you look at https://github.com/GoogleCloudPlatform/magic-modules/pull/688, I encountered + resolved the same issue.

Reply:
hmmm, @rileykarson since this is a generated resource, is that an MM bug that needs to be addressed?

Reply:
Yep - I talked over chat with @drebes about this briefly, and after mulling over it afterwards I think we should solve this in MM.

We can probably break out the schema + explicitly set every `TypeSet`, that way these nested ones will work and we'll be using the same logic in every case.

Reply:
I'll look into the general MM solution for this!

Reply:
Thanks, I'll wait for the fix then. In the meanwhile I'll try to add support for DNS forwarding which should be in public beta around these days.